sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1891,48 +1916,88 @@ object RapidsConf {
+        // scalastyle:off line.size.limit
+        println("""# RAPIDS Accelerator for Apache Spark Configuration
+          |The following is the list of options that `rapids-plugin-4-spark` supports.
+          |
+          |On startup use: `--conf [conf key]=[conf value]`. For example:
+          |
+          |```
+          |${SPARK_HOME}/bin/spark-shell --jars rapids-4-spark_2.12-23.06.0-SNAPSHOT-cuda11.jar \
+          |--conf spark.plugins=com.nvidia.spark.SQLPlugin \
+          |--conf spark.rapids.sql.concurrentGpuTasks=2
+          |```
+          |
+          |At runtime use: `spark.conf.set("[conf key]", [conf value])`. For example:
+          |
+          |```
+          |scala> spark.conf.set("spark.rapids.sql.concurrentGpuTasks", 2)
+          |```
+          |
+          | All configs can be set on startup, but some configs, especially for shuffle, will not
+          | work if they are set at runtime. Please check the column of "Applicable at" to see
+          | when the config can be set. "Startup" means only valid on startup, "Runtime" means
+          | valid on both startup and runtime.
+          |""".stripMargin)
+        // scalastyle:on line.size.limit
+        println("\n## General Configuration\n")
+      } else {
+        // print advanced configuration
+        println("title: Advanced Configuration")
+        println("parent: Additional Functionality")
+        println("nav_order: 10")
+        println("---")
+        println(s"<!-- Generated by RapidsConf.help. DO NOT EDIT! -->")
+        // scalastyle:off line.size.limit
+        println("""# RAPIDS Accelerator for Apache Spark Advanced Configuration
+          |Most users will not need to modify the configuration options listed below. 
+          |They are documented here for completeness and advanced usage.
+          |
+          |The following configuration options are supported by the RAPIDS Accelerator for Apache Spark.
+          |
+          |For commonly used configurations and examples of setting options, please refer to the 
+          |[RAPIDS Accelerator for Configuration](https://nvidia.github.io/spark-rapids/docs/configs.html) page.
+          |""".stripMargin)
+        // scalastyle:on line.size.limit
+        println("\n## Advanced Configuration\n")
+      }
 
-      println("\n## General Configuration\n")
       println("Name | Description | Default Value | Applicable at")
       println("-----|-------------|--------------|--------------")
     } else {
       println("Rapids Configs:")
     }
     val allConfs = registeredConfs.clone()
     allConfs.append(RapidsPrivateUtil.getPrivateConfigs(): _*)
-    allConfs.sortBy(_.key).foreach(_.help(asTable))
+    val outputConfs = commonlyUsed match {
+      case Some(true) => allConfs.filter(_.isCommonlyUsed)
+      case Some(false) => allConfs.filter(!_.isCommonlyUsed)
+      case None => allConfs
+    }
+    outputConfs.sortBy(_.key).foreach(_.help(asTable))
+    if (commonlyUsed.get == true) {
```

Comment:
what if `commonlyUsed` is None ?

Reply:
It will print all the configs as we used the help function before.

Reply:
thanks, removed it.